ndc-models/src/lib.rs

```diff
@@ -420,9 +420,37 @@ pub struct Query {
     pub offset: Option<u32>,
     pub order_by: Option<OrderBy>,
     pub predicate: Option<Expression>,
+    ///
+    pub groups: Option<Grouping>,
 }
 // ANCHOR_END: Query
 
+// ANCHOR: Grouping
+#[skip_serializing_none]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Grouping")]
+pub struct Grouping {
+    /// Dimensions along which to partition the data
+    pub dimensions: IndexMap<String, Dimension>,
```

Comment:
I'm concerned about using a JSON object to represent this structure. The ordering of dimensions is potentially important (see Rollup grouping type in the RFC; yes this is out of scope for now, but let's not ignore the potential) and JSON objects don't care about property ordering.

Reply:
`IndexMap` does preserve the order of keys, so you can consider it as an ordered map structure. But maybe it's better to change this to a vector. I would then also remove the names from the query request.

Reply:
Note that the order of fields in the query response potentially is also important, which is why we use `IndexMap` (for GraphQL at least, it should match the requested order)

Reply:
My issue is less about the data structure chosen in Rust, and more that it translates to a JSON object in the JSON Schema. JSON Schema doesn't say anything about object property ordering being retained afaik. And if other programming languages (cough, Haskell with `aeson`) consume JSON Objects they often shove them in hashmaps and the ordering is lost.

If the ordering is important, we should serialize to an array in JSON IMHO, so that's explicit.

Reply:
Agreed, I've changed the dimensions list to be a vector.